full/Dockerfile

```diff
@@ -1,192 +1,149 @@
+        | tar -xzvC $MAVEN_HOME --strip-components=1
+
+### PHP ###
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -yq \
         composer \
+        php \
         php-all-dev \
-        php-intl \
-        php-mysql \
-        php-pgsql \
-        php-xml \
+        php-curl \
+        php-date \
         php-gd \
+        php-gettext \
+        php-intl \
         php-json \
-        php-curl \
         php-mbstring \
+        php-mysql \
+        php-net-ftp \
+        php-pgsql \
         php-sqlite3 \
         php-tokenizer \
-        php-ctype \
-        php-date \
-        php-net-ftp \
-        php-gettext \
         php-zip \
-    && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
- # the PHP language server is installed by the theia-php-extension
+    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/*
+# PHP language server is installed by theia-php-extension
+
+### Yarn ###
+RUN curl -fsSL https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
+    && apt-add-repository -yu "deb https://dl.yarnpkg.com/debian/ stable main" \
+    && apt-get install --no-install-recommends -yq yarn=1.12.3-1 \
+    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/*
+
+### Gitpod user (2) ###
+USER gitpod
+# use sudo so that user does not get sudo usage info on (the first) login
+RUN sudo echo "Running 'sudo' for Gitpod: success"
+
+### Go ###
+ENV GO_VERSION=1.11.2 \
+    GOPATH=$HOME/go-packages \
+    GOROOT=$HOME/go
+ENV PATH=$GOROOT/bin:$GOPATH/bin:$PATH
+RUN curl -fsSL https://storage.googleapis.com/golang/go$GO_VERSION.linux-amd64.tar.gz | tar -xzv \
+    && go get -u -v \
+        github.com/acroca/go-symbols \
+        github.com/cweill/gotests/... \
+        github.com/davidrjenni/reftools/cmd/fillstruct \
+        github.com/fatih/gomodifytags \
+        github.com/haya14busa/goplay/cmd/goplay \
+        github.com/josharian/impl \
+        github.com/nsf/gocode \
+        github.com/ramya-rao-a/go-outline \
+        github.com/rogpeppe/godef \
+        github.com/uudashr/gopkgs/cmd/gopkgs \
+        github.com/zmb3/gogetdoc \
+        golang.org/x/lint/golint \
+        golang.org/x/tools/cmd/godoc \
+        golang.org/x/tools/cmd/gorename \
+        golang.org/x/tools/cmd/guru \
+        sourcegraph.com/sqs/goreturns
+# user Go packages
+ENV GOPATH=/workspace:$GOPATH \
+    PATH=/workspace/bin:$PATH
+
+### Node.js ###
+ARG NODE_VERSION=8.14.0
+ENV PATH=/home/gitpod/.nvm/versions/node/v8.14.0/bin:$PATH
+RUN curl -fsSL https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | bash \
+    && bash -c ". .nvm/nvm.sh \
+        && npm config set python /usr/bin/python --global \
+        && npm config set python /usr/bin/python \
+        && npm install -g typescript"
+
+### Python ###
+ENV PYENV_ROOT=$HOME/.pyenv
+ENV PATH=$PYENV_ROOT/bin:$PATH
```

Comment:
this used to be `PATH="$HOME/.pyenv/bin:$HOME/.pyenv/shims:$PATH"`. Is `shims` still on the `PATH` or should it not be?

Reply:
It's not there anymore, and I wanted to ask whether it was useful for anything. My general impression was that some of the variables were defined only to be used in the Dockerfile (before we discovered `ARG`), or so that the installation instructions in some of the following lines would work, or could be written more concisely (i.e. using just the basename as they would be in a path already).

Reply:
Oh no, I didn't want to remove them, I just noticed they're added to the `$PATH` by a piece of code the installer adds to `.bashrc`. So I was thinking that we shouldn't duplicate what the installer does (in order to avoid a potential mismatch, not mentioning that we'd have it in two copies in the `$PATH`), that was the reason why I removed this. But unfortunately such a piece of code gets executed just sometimes (only in Bash (as opposed to `sh`) and depending on the combination of switches `-i` and `-l` you use). So it's a bit tricky situation - as I mentioned above. I'll double-check before merging, if unsure, I'll put the explicit manual setting of $PATH back, OK?

Reply:
Ok. Just saying, this need to be tested carefully. Python language support depends on that. I'm quite sure we're using `bash` as the shell when we ssh-ing into the workspace, so the bashrc-only solution should work.